File: programmers/python 3/12945.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12945

def solution(n):
    F = [0, 1]
    for i in range(2, n + 1):
        F.append(F[i - 2] + F[i - 1])
    return F[n] % 1234567

File: programmers/swift/12945.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12945

func solution(_ n:Int) -> Int {
    var a = 0
    var b = 1
    for _ in 2...n {
        (a, b) = (b, (a + b) % 1234567)
    }
    return b
}